### ISSUE TITLE
Unintended forces replacement in google_vpc_access_connector

### ISSUE THREAD

**Issue by sho-abe** [hashicorp/terraform-provider-google#12411]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
Terraform v1.2.8
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.33.0
+ provider registry.terraform.io/hashicorp/google-beta v4.33.0
```

### Affected Resource(s)

* google_vpc_access_connector

### Terraform Configuration Files

```tf
resource "google _compute_network" "main" {
  project                 = "your-project-id"
  name                    = "vpc-access-test"
  auto_create_subnetworks = false
  routing_mode            = "REGIONAL"
}

resource "google_compute_subnetwork" "main" {
  project       = "your-project-id"
  ip_cidr_range = "10.10.0.0/28"
  name          = "vpc-access-connector-sub"
  network       = google_compute_network.main.self_link
  region        = "asia-northeast1"
}

resource "google_vpc_access_connector" "connector" {
  provider = google-beta
  project  = "your-project-id"
  name     = "vpc-access-connector"
  region   = google_compute_subnetwork.main.region

  subnet {
    name = google_compute_subnetwork.main.name
  }

  machine_type  = "e2-micro"
  min_instances = 5
  max_instances = 10
}
```

### Expected Behavior

No difference occurs in google_vpc_access_connector when executing `terraform plan/apply`.

### Actual Behavior

Forces replacements occur in google_vpc_access_connector when executing `terraform plan/apply`.

### Steps to Reproduce

1. Create a tf file as per the "Terraform Configuration Files" section.
2. Create VPC Access connector by `terraform apply`.
3. Re-execute `terraform plan` or `terraform apply`.

### Important Factoids

The default values are set in the google-beta provider for `max_throughput` and `min_throughput`.
It seems that these default values unintentionally conflict with the `max_instances` and `min_instances`.

* https://github.com/hashicorp/terraform-provider-google-beta/blob/94460d31d6619a073687b08e7502052f60a9cba5/google-beta/resource_vpc_access_connector.go#L76
* https://github.com/hashicorp/terraform-provider-google-beta/blob/94460d31d6619a073687b08e7502052f60a9cba5/google-beta/resource_vpc_access_connector.go#L91

### References

* https://github.com/hashicorp/terraform-provider-google-beta/blob/v4.33.0/google-beta/resource_vpc_access_connector.go

**Comment by edwardmedia** [comment]
Looks like the defaults are
```
    "minThroughput": 500,
    "maxThroughput": 1000,
```

**Comment by sho-abe** [comment]
@edwardmedia Thank you for your response.
I think there is not a good way to change Terraform provider's default values.
Because the difference is not detected when `max_instances = 2` and `min_instances = 3` as below.

```tf
resource "google_vpc_access_connector" "connector" {
  provider = google-beta
  name     = "vpc-access-connector"
  region   = google_compute_subnetwork.main.region

  subnet {
    name = google_compute_subnetwork.main.name
  }

  machine_type  = "e2-micro"
  min_instances = 2
  max_instances = 3
}
```

I guess the API is also changing the `max_throughput` and `min_throughput` values, probably according to the `max_instances` and `min_instances` values.
So I think it's better not to set default values of `max_throughput` and `min_throughput` in the Terraform provider.

Regards

**Comment by melinath** [comment]
previous PR caused test failures that represent real user issues - see https://github.com/GoogleCloudPlatform/magic-modules/pull/6541 for revert PR.

**Comment by sho-abe** [comment]
It looks to me like this issue has been fixed in version 6.0.0.

https://github.com/hashicorp/terraform-provider-google/releases/tag/v6.0.0
> vpcaccess: removed default values for min_throughput and min_instances fields on google_vpc_access_connector and made them default to values returned from the API when not provided by users (https://github.com/hashicorp/terraform-provider-google/pull/18697)

Thanks,

**Comment by melinath** [comment]
great to hear, thanks for catching that!
